src/server/query_processor.py

```diff
@@ -50,15 +39,13 @@ async def process_query(
         Type of pattern to use (either "include" or "exclude") (default: ``"exclude"``).
     pattern : str
         Pattern to include or exclude in the query, depending on the pattern type.
-    is_index : bool
-        Flag indicating whether the request is for the index page (default: ``False``).
     token : str | None
         GitHub personal access token (PAT) for accessing private repositories.
 
     Returns
     -------
-    _TemplateResponse
-        Rendered template response containing the processed results or an error message.
+    dict[str, Any]
```

Comment:
Let's use something else than `Any` here. At least use `object` for now.

```suggestion
    dict[str, object]
```

And remove the import statement of `Any`.

Reply:
Maybe define something like this:

```python
from pydantic import BaseModel, Field, ConfigDict


class IngestResult(BaseModel):

    summary: str
    tree: str
    content: str

    model_config = ConfigDict(extra="forbid")


class QueryContext(BaseModel):

    repo_url: str
    default_file_size: int
    pattern_type: str
    pattern: str
    short_repo_url: str | None
    token: str | None
    examples: list[str]
    error: str | None
    error_message: str | None
    result: IngestResult | None

    model_config = ConfigDict(extra="forbid")
```

I see that you are already defining something similar. Maybe there is some overlap. `IngestSuccessResponse`, `IngestErrorResponse` and `IngestResponse = Union[IngestSuccessResponse, IngestErrorResponse]`.